plugins/inputs/modbus/request.go

```diff
@@ -53,16 +57,23 @@ func groupFieldsToRequests(fields []field, tags map[string]string, maxBatchSize
 			current.length += f.length
 			if !f.omit {
 				// Omit adding the field but use it for constructing the request.
-				current.fields = append(current.fields, f)
+				if isNonEmptyRequest(current) {
+					current.fields = append(current.fields, f)
+				} else {
+					current = newRequest(f, tags)
+				}
```

Comment:
How about taking another approach here? For the "starting with omit" case we could simply modify `newRequest` to omit the field like
```golang
func newRequest(f field, tags map[string]string) request {
	r := request{
		address: f.address,
		length:  f.length,
		fields:  []field{},
		tags:    map[string]string{},
	}
	if !f.omit {
		r.fields = append(r.fields, f)
	}

	// Copy the tags
	for k, v := range tags {
		r.tags[k] = v
	}
	return r
}
```
and later check the requests. See below.